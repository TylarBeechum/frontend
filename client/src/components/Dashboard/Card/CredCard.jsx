import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CredCardViewBtn from './CredCardViewBtn';
import CredCardSchoolName from './CredCardSchoolName';
import CredCardDateIssued from './CredCardDateIssued';
import CredCardStudentName from './CredCardStudentName';
import CredCardDeleteBtn from './CredCardDeleteBtn';

const CredCard = ({ cred }) => {
  const { credName, criteria, ownerName } = cred;
  return (
    <CredContainer>
      <CredCardViewBtn />
      <CredCardSchoolName credName={credName} criteria={criteria} />
      <CredCardDateIssued />
      <CredCardStudentName ownerName={ownerName} />
      <CredCardDeleteBtn />
    </CredContainer>
  );
};

CredCard.propTypes = {
  cred: PropTypes.shape({
    credName: PropTypes.string,
    criteria: PropTypes.string,
    ownerName: PropTypes.string
  })
};

CredCard.defaultProps = {
  cred: {
    credName: '',
    criteria: '',
    ownerName: ''
  }
};

const CredContainer = styled.section`
  margin: 10px auto 0;
  max-width: 1600px;
  width: 100%;
  background: white;
  /* min-height: calc(100vh - 170px); */
  -webkit-box-shadow: -2px 5px 25px -17px rgba(0, 0, 0, 0.61);
  -moz-box-shadow: -2px 5px 25px -17px rgba(0, 0, 0, 0.61);
  box-shadow: -2px 5px 25px -17px rgba(0, 0, 0, 0.61);
  border-radius: 2px;
  display: flex;
  /* justify-content: space-between; */
  padding: 20px 0;
  transition: box-shadow 0.5s;

  :hover {
    -webkit-box-shadow: 0px 0px 15px -2px rgba(125, 76, 219, 1);
    -moz-box-shadow: 0px 0px 15px -2px rgba(125, 76, 219, 1);
    box-shadow: 0px 0px 15px -2px rgba(125, 76, 219, 1);
  }
`;

export default CredCard;
