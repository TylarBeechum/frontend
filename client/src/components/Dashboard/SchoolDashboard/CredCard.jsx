import React from 'react';
import styled from 'styled-components';
import CredCardViewBtn from './CredCardViewBtn';
import CredCardSchoolName from './CredCardSchoolName';
import CredCardDateIssued from './CredCardDateIssued';
import CredCardStudentName from './CredCardStudentName';
import CredCardDeleteBtn from './CredCardDeleteBtn';

const CredCard = ({ credName }) => {
  return (
    <CredContainer>
      <CredCardViewBtn />
      <CredCardSchoolName credName={credName} />
      <CredCardDateIssued />
      <CredCardStudentName />
      <CredCardDeleteBtn />
    </CredContainer>
  );
};

const CredContainer = styled.section`
  margin: 10px auto 0;
  max-width: 1600px;
  width: 100%;
  background: white;
  /* min-height: calc(100vh - 170px); */
  border: 1px solid #d8d8d8;
  border-radius: 2px;
  display: flex;
  justify-content: space-between;
  padding: 20px 0;
`;

export default CredCard;
