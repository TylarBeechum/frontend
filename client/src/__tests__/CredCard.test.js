/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import CredCard from '../components/Dashboard/Card/CredCard';

describe('<CredCard />', () => {
  it('should render the CredCard component without crashing', () => {
    const cred = {
      credName: 'test',
      credHash: 'string',
      criteria: 'testing',
      ownerName: 'Stampd',
      id: '5'
    };
    const helpers = renderWithRouterAndProviders(<CredCard cred={cred} />);
  });

  it('matches the snapshot of App', () => {
    const cred = {
      credName: 'test',
      credHash: 'string',
      criteria: 'testing',
      ownerName: 'Stampd',
      id: '5'
    };
    const tree = rendererWithRouterAndProviders(
      <CredCard cred={cred} history={{ location: { pathname: '/' } }} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
