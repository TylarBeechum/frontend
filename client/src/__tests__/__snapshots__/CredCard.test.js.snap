// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CredCard /> matches the snapshot of App 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <section
    className="sc-VigVT kwgGCs"
  >
    <div
      className="sc-EHOje hWneJQ"
    >
      <button
        aria-label="view credentials button"
        className="sc-bdVaJa sc-bZQynM csIfUH StyledButton-sc-323bzc-0 kHrfdt"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        View
      </button>
    </div>
    <div
      className="sc-gzVnrw bKsZFP"
    >
      <h2>
        test
      </h2>
      <p>
        testing
      </p>
    </div>
    <div
      className="sc-htoDjs erhmUl"
    >
      <p />
    </div>
    <div
      className="sc-dnqmqq jubHKb"
    >
      <h3>
        Stampd
      </h3>
    </div>
  </section>
</div>
`;
